### ISSUE TITLE
Fails to initialize CUE SDK

### ISSUE THREAD

**Issue by NikWillOrStuff** [antonpup/Aurora#724]
### What are you experiencing an issue with?

- [ ] Aurora
- [ ] Aurora supported game
- [ ] Aurora Wrapper
- [ ] Aurora Profile
- [x] Aurora supported device, (Corsair K95 RGB, CUE 1.16.42)
- [ ] Aurora Scripts
- [ ] Aurora Updater
- [ ] Aurora Lighting issue
- [ ] Other, (Fill in another issue here)

### What kind of an issue is it?

- [ ] Crash
- [ ] Performance problems
- [ ] Visual bug
- [ ] Localization issue
- [ ] Enhancement suggestion
- [x] Device issue
- [ ] Other, (Fill in another type of issue here)

### Expected behavior:
should start doing fancy RGB stuffs on my keyboard

### Actual behavior:
fails to initialize corsair SDK thingy

### How often does this problem occur?

I haven't gotten it to correctly connect once yet.

### Please describe the steps to reproduce this issue:
The issue can be reproduced by doing following steps:

1. open CUE 1.16.42, enable SDK
2. open aurora
3. go to device manager tab in settings
4. click "Start" button on the corsair thing and watch nothing happen

### What is the version of Aurora you are experiencing this issue with?

v0.6.1-dev2 and beta v0.6.0b

### Please include any logs, screenshots, or files that are related to this issue:

here's the log
https://pastebin.com/sAuqarPn

### Extra information
This was working previously, on the same computer with the same version of CUE and windows. But I had to reinstall windows and set everything up again, and now aurora isn't working this time. I don't know the exact version of aurora I was using previously, but I do have this screenshot http://i.imgur.com/qxZVOcT.png

**Comment by simon-wh** [comment]
Could you update CUE? Because currently CUE is on version 2 and the source of your issues might be due to your CUE version being too old.

**Comment by NikWillOrStuff** [comment]
I had it working with this version of CUE before and I prefer it because of it's better control over macros etc. also the prompts for enabling the CUE SDK in aurora are still CUE 1 so I thought it was better supported. but I just uninstalled CUE 1 and installed CUE 2 and it still didn't work. in fact the log looks identical https://pastebin.com/8Qcn2TWa

**Comment by simon-wh** [comment]
@NikWillOrStuff Is the SDK definitely enabled when you're on CUE 2? If it is, I can't think of any other reason why it wouldn't work, you can see over at [this issue](https://github.com/DarthAffe/CUE.NET/issues/55) on CUE.NET that the lib developer doesn't know either as this seems like more of a CUE.NET issue rather than Aurora. However, I'm going to try and setup a debug version of CUE.NET so I can try and pinpoint the source of the issue.

**Comment by NikWillOrStuff** [comment]
yeah, I made sure the SDK was enabled for CUE 1 and 2. I'd be happy to do any tests you want :)

**Comment by Dagda52** [comment]
@simon-wh now I got this error too after uninstalling stable version and trying 0.6.1 dev 2, only works if pc is restarted, but fails to re hook if i close aurora and try open it again or just open another third party software that uses cue sdk here is my log
https://pastebin.com/n30kYJc9

**Comment by simon-wh** [comment]
@Dagda52 From your log it shows that the launch directory is at one point is SysWOW64 and then the CUE program files directory. The launch directory needs to be the directory that Aurora is in for CUE.NET to find the sdk dlls, how are you launching aurora?

**Comment by Dagda52** [comment]
@simon-wh i made a folder called aurora in the desktop, and launching from there, but i made also a shortcut  to one of my g keys on my keyboard, it did't had any trouble before, but even so it happens anyways if i lauch directly from the exe inside the folder in desktop or the g key.

**Comment by simon-wh** [comment]
Well, I've made it so it forces setting the launch directory which should solve the issue for the next release

**Comment by brainbug89** [comment]
@Dagda52 
> https://pastebin.com/n30kYJc9

Line 884 and 885 from your log file:
´´´
'C:\Program Files (x86)\Corsair\Corsair Utility Engine\x64\CUESDK_2015.dll
C:\Program Files (x86)\Corsair\Corsair Utility Engine\x64\CUESDK.dll'
´´´

This folder doesn't exist on my computer. And I don't get this error in my log files. This error gets generated at (https://github.com/DarthAffe/CUE.NET)[CUE.NET]
The error gets thrown (https://github.com/DarthAffe/CUE.NET/blob/c786710b65e557e1ed1271a5ff3ea74dd0414a4a/Native/_CUESDK.cs#L51)[here]. The path should be the **Aurora folder** and not ´C:\Program Files (x86)\Corsair\Corsair Utility Engine\´


> but even so it happens anyways if i lauch directly from the exe inside the folder

If you launch directly from the exe inside the Aurora folder, CUE.NET should find the .dll :thinking:

**Comment by Dagda52** [comment]
apparently is fixed now in dev 3 at least for me

**Comment by NikWillOrStuff** [comment]
I installed v0.6.1-dev3 from the releases on github (the webinstaller installed a version of aurora that instantly crashed, I posted more info [here](https://github.com/antonpup/Aurora/issues/824#issuecomment-334561712)). and I don't have this issue anymore! but what's particularly weird, I uninstalled dev3 and installed v0.6.0-dev2 (it says "beta v0.6.1-dev2" in aurora itself) again, which is what I tested with when I submitted this bug report, and it *also* works now. I did not update CUE or reinstall my OS, though I've no doubt done some windows updates as it's been a few months (sorry!)
